spec/lib/pdf_filler/nj1040_pdf_spec.rb

```diff
@@ -279,6 +279,41 @@
       end
     end
 
+    describe "esign date" do
+      context 'filer is single' do
+        let(:submission) {
+          create :efile_submission, tax_return: nil, data_source: create(
+            :state_file_nj_intake,
+            primary_ssn: "123456789",
+            primary_esigned: "yes",
+            primary_esigned_at: DateTime.new(2024, 12, 19, 12)
```

Comment:
A good question, thanks for bringing this up! In the context of this PR, we are adding tests for PDF and XML generation, so I'm thinking through them individually below. The DateTime Ruby docs show it's possible to specify a timezone for DateTime objects: https://ruby-doc.org/stdlib-2.6.1/libdoc/date/rdoc/DateTime.html#method-c-new. It defaults to UTC, which means we're currently creating DateTime with an offset of 0. 

XML Generation Tests (ALL STATES)
The DateTime object is converted to a string by this helper method:

https://github.com/codeforamerica/vita-min/blob/main/app/lib/submission_builder/formatting_methods.rb#L29

Which converts a DateTime into a date using state_code.

The existing tests use ET explicitly when checking for midnight edge cases (see lines 288 and 307 in `return_header_spec.rb`), but it's helpful to note that the max offset for the contingous 48 states in the US is UTC-8 (Hawaii is UTC-10 and Alaska is UTC-9). In all those cases, noon UTC will still be the same date when converted to local state_code date. There should be no date discrepancy here.

PDF Generation Tests (NJ ONLY)
The PDF generation takes the XML string and compares it to a static expected string, 2024-12-19. The converted datetime will always be on this day for NJ since we're converting the noon 12-19-2024 UTC timestamp to ET (UTC-5). This means that it corresponds to 12-19-2024, 7am.

Hope this answers your question, let me know if you have additional concerns.

Reply:
I'm going to rewrite the return_header_spec tests to test all states DateSigned functionality in the next commit

Reply:
This is proving tougher to test than expected, given that every state has a different way of indicating that a spouse exists 😬 . 

Ideally, I'd create an intake for a single filer, verify that the XML has the correct `Primary DateSigned` attribute present, and no `Secondary DateSigned` element. Then I'd create an intake for a filer with a spouse, verify the XML has correct `Primary DateSigned` and `Secondary DateSigned` attributes.

Suggestions for how to set the spouse case across all active states? It seems right now like I'd have to modify every state's intake factory to set `spouse_esigned` and `spouse_esigned_at`. I could do that in the default case, which would make things easy, but it seems very wrong to set `spouse_esigned` and `spouse_esigned_at` for the default single filer.